src/Serializer/Filter/GroupFilter.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Serializer\Filter;
+
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Group filter.
+ *
+ * @author Baptiste Meyer <baptiste@les-tilleuls.coop>
+ */
+class GroupFilter implements FilterInterface
+{
+    protected $overrideDefaultGroups;
+    protected $parameterName;
+
+    public function __construct(string $parameterName = 'groups', bool $overrideDefaultGroups = false)
+    {
+        $this->overrideDefaultGroups = $overrideDefaultGroups;
+        $this->parameterName = $parameterName;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function apply(Request $request, bool $normalization, array $attributes, array &$context)
+    {
+        if (empty($groups = $this->extractGroups($request))) {
+            return;
+        }
+
+        if (!$this->overrideDefaultGroups && !empty($context['groups'])) {
+            $groups = array_merge($context['groups'], $groups);
+        }
+
+        $context['groups'] = $groups;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDescription(string $resourceClass): array
+    {
+        return [
+            $this->parameterName => [
+                'property' => null,
+                'type' => 'string',
+                'required' => false,
+            ],
+            $this->parameterName.'[]' => [
+                'property' => null,
+                'type' => 'string',
+                'required' => false,
+            ],
+        ];
+    }
+
+    /**
+     * Extracts groups from a request.
+     *
+     * @param Request $request
+     *
+     * @return array
+     */
+    protected function extractGroups(Request $request): array
+    {
+        $groups = [];
+
+        if (null !== $groupsParameter = $request->query->get($this->parameterName)) {
+            if (is_string($groupsParameter)) {
```

Comment:
I don't think we should `explode` or do any transformation on query parameters. Just let symfony do the work here.

I'd propose:

```php
$groups = $request->query->get($this->parameterName);
```

This method can then be removed as this one line should be enough.